pandas/tests/io/formats/test_format.py

```diff
@@ -3240,14 +3242,30 @@ def test_repr_html_ipython_config(ip):
 
 
 @pytest.mark.parametrize("method", ["to_string", "to_html", "to_latex"])
+@pytest.mark.parametrize("encoding", [None, "utf-8", "gbk", "foo"])
 def test_filepath_or_buffer_arg(
-    float_frame, method, filepath_or_buffer, assert_filepath_or_buffer_equals
+    float_frame,
+    method,
+    filepath_or_buffer,
+    assert_filepath_or_buffer_equals,
+    encoding,
+    filepath_or_buffer_id,
 ):
     df = float_frame
```

Comment:
Is there a reason to use `float_frame` here? I think can just construct a DataFrame using the data you have on line below instead of modifying this; as is its not clear on intent to use this fixture

Reply:
@simonjayhawkins `float_frame` fixture has been uniformly used in all the tests, hence it is used here.

Reply:
maybe.. 

```diff
diff --git a/pandas/tests/io/formats/test_format.py b/pandas/tests/io/formats/test_format.py
index 096fc6cb4..dc2784a7b 100644
--- a/pandas/tests/io/formats/test_format.py
+++ b/pandas/tests/io/formats/test_format.py
@@ -3240,16 +3240,19 @@ def test_repr_html_ipython_config(ip):
 
 
 @pytest.mark.parametrize("method", ["to_string", "to_html", "to_latex"])
-@pytest.mark.parametrize("encoding", [None, "utf-8", "gbk", "foo"])
+@pytest.mark.parametrize(
+    "encoding, data",
+    [(None, "abc"), ("utf-8", "abc"), ("gbk", "造成输出中文显示乱码"), ("foo", "abc")],
+)
 def test_filepath_or_buffer_arg(
-    float_frame,
     method,
     filepath_or_buffer,
     assert_filepath_or_buffer_equals,
     encoding,
+    data,
     filepath_or_buffer_id,
 ):
-    df = float_frame
+    df = DataFrame([data])
 
     if filepath_or_buffer_id not in ["string", "pathlike"] and encoding is not None:
         with pytest.raises(
```

Reply:
Looks fine, I'll make the changes and push.